pages/api/preview.js

```diff
@@ -0,0 +1,10 @@
+export default function handler(req, res) {
+  const isPreview = req.preview;
+
+  if (isPreview) {
+    res.clearPreviewData();
+  } else {
+    res.setPreviewData({});
+  }
+  res.redirect("/");
```

Comment:
@logan-anderson says there's a clean way to grab the referrer URL and send the user back to it 👍